src/test/java/com.gigssandbox/services/UserServiceTest.java

```diff
@@ -0,0 +1,93 @@
+package com.gigssandbox.services;
+
+import com.gigssandbox.entities.User;
+import com.gigssandbox.exceptions.AlreadyLoggedInException;
+import com.gigssandbox.exceptions.AlreadyRegisteredException;
+import com.gigssandbox.exceptions.IncorrectPasswordException;
+import com.gigssandbox.exceptions.NotRegisteredException;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class UserServiceTest {
+    private UserService userService;
+    private String username;
+    private String password;
+    private User user;
+    private Map<String, User> users;
+
+    @Before
+    public void setUp() {
+        this.username = "oliver_sykes";
+        this.password = "poprockisthebest";
+        this.user = User
+                .builder()
+                .username(username)
+                .passwordHash(password.hashCode())
+                .build();
+        this.users = new HashMap<>();
+
+        this.userService = new UserService(users);
+    }
+
+    @Test
+    public void shoudAddUserToUsersWhenUserTriesToRegister() throws Exception {
+        userService.registerUser(username, password);
+
+        assertTrue(userService.exists(username));
```

Comment:
userService.exists() is used only for testing. You've already added DI. Check injected map contains user  
```
assertTrue(users.contains(username));
```